warehouse/query-core/src/main/java/datawave/query/jexl/visitors/EdgeTableRangeBuildingVisitor.java

```diff
@@ -168,75 +171,86 @@ public Object visit(ASTAndNode node, Object data) {
-                qContext.packageIdentities((List<IdentityContext>) contexts1);
-            }
-            
-            return contexts2;
-        } else if ((contexts1.get(0) instanceof QueryContext) && (contexts2.get(0) instanceof IdentityContext)) {
-            
-            for (QueryContext qContext : (List<QueryContext>) contexts1) {
-                qContext.packageIdentities((List<IdentityContext>) contexts2);
-            }
+        List<? extends EdgeContext> mergedContext = childContexts.remove(childContexts.size() - 1);
+        
+        // now merge the child contexts
+        while (!childContexts.isEmpty()) {
+            List<? extends EdgeContext> childContext = childContexts.remove(childContexts.size() - 1);
             
-            return contexts1;
-            /*
-             * On rare occasion a group of Query contexts without a source can get grouped together, this happens with queries like: SOURCE == 's1' && ((TYPE ==
-             * 't1' && RELATIONSHIP == 'r1') || (TYPE == 't2' && RELATIONSHIP == 'r2'))
-             * 
-             * This probably was not supposed to be allowed, you should only be ANDing groups of sources with groups of other identifiers rather here it is
-             * ANDing source(s) with groups of expressions. Honestly it would be safer to split that type of query up into two separate ones but limited support
-             * has been included. However it can be dangerous if the user tries to use SINK in one of the grouped expressions so that is not allowed.
-             */
-        } else if ((contexts1.get(0) instanceof QueryContext) && (contexts2.get(0) instanceof QueryContext)) {
-            // Assumes that if the first query context does not have a row context then they all don't
-            if (((List<QueryContext>) contexts1).get(0).getRowContext() != null) {
-                // The size of the list for contexts1 is usually going to be 1
-                for (QueryContext qContext : ((List<QueryContext>) contexts1)) {
-                    // Combine the query contexts if anything fails blame the user
-                    if (!(qContext.combineQueryContexts(((List<QueryContext>) contexts2), false))) {
-                        log.error("And node had unexpected return type");
-                        throw new IllegalArgumentException("Error: problem with query syntax");
-                    }
+            if ((childContext.get(0) instanceof IdentityContext) && (mergedContext.get(0) instanceof IdentityContext)) {
+                QueryContext qContext = new QueryContext();
+                
+                qContext.packageIdentities((List<IdentityContext>) childContext);
+                qContext.packageIdentities((List<IdentityContext>) mergedContext);
+                
+                ArrayList<QueryContext> aList = new ArrayList<>();
+                aList.add(qContext);
+                mergedContext = aList;
```

Comment:
how about set "processed = true" as well and then test it at the error message below.

Reply:
Check out my latest commit.  I think that by putting the thrown exception in an else statement we fix the issue and preserve the original functionality.